src/main/java/mezz/jei/gui/recipes/RecipeTransferButton.java

```diff
@@ -33,7 +33,7 @@ public void init(RecipeTransferManager recipeTransferManager, @Nullable Containe
 			this.recipeTransferError = RecipeTransferErrorInternal.INSTANCE;
 		}
 
-		if (this.recipeTransferError == null) {
+		if (this.recipeTransferError == null || this.recipeTransferError.getType().equals(IRecipeTransferError.Type.COSMETIC)) {
```

Comment:
I'd like this to be pulled out into a helper function like `allowsTransfer(@nullable recipeTransferError)` that returns true or false. Then you can re-use it and make these `if` statements more readable.

Also, I think you can safely use `==` for the enum comparison here

Reply:
Good idea. I was struggling a lot with those ifs 😅